runtime/near-evm-runner/src/utils.rs

```diff
@@ -183,66 +187,497 @@ pub fn format_log(topics: Vec<H256>, data: &[u8]) -> std::result::Result<Vec<u8>
     Ok(result)
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Type {
+    Address,
+    Uint,
+    Int,
+    String,
+    Bool,
+    Bytes,
+    Byte(u8),
+    Custom(String),
+    Array { length: Option<u64>, inner: Box<Type> },
+}
+
+/// the type string is being validated before it's parsed.
+pub fn parse_type(field_type: &str) -> Result<Type> {
```

Comment:
What's the input format?

Reply:
A single evm function arg type in string, without the argument name. e.g. `"bytes"` `"uint256[][3]"` `"CustomStructName"`